admin-ui/src/components/rules/rules.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import './rules.css';
+import { Row, Col, Icon } from 'antd';
+import { Controlled as CodeMirror } from 'react-codemirror2';
+import 'codemirror/theme/material.css';
+import 'codemirror/lib/codemirror.css';
+
+class Rules extends Component {
+	constructor(props) {
+		super(props);
+		const rules = Object.keys(this.props.rules);
+		var selectedRule = '';
+		if (rules.length > 0) {
+			selectedRule = rules[0];
+		}
+		this.state = { selectedRule: selectedRule };
+		this.handleChange = this.handleChange.bind(this);
+		this.handleClick = this.handleClick.bind(this);
+	}
+	handleChange(editor, data, value) {
+		console.log(value);
+		this.props.handleRuleChange(value);
+	}
+	handleClick(rule) {
+		this.setState({ selectedRule: rule });
+	}
+	render() {
+		var rules = Object.keys(this.props.rules);
+		const values = Object.values(this.props.rules);
```

Comment:
There is no need of making values and index here since you can directly access value of a particular rule by this.props.rules[this.state.selectedRule]

Reply:
Not resolved yet. You are still using values array